assets/js/components/Dashboard/PaMessaging/NePaMessagePage/MessageCard.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { ComponentType } from "react";
+import { Card, Row, Col, Form, Button } from "react-bootstrap";
+import { ArrowRightShort, VolumeUpFill } from "react-bootstrap-icons";
+import {
+  NewPaMessagePageState,
+  NewPaMessagePageReducerAction,
+} from "./NewPaMessagePage";
+
+interface Props {
+  pageState: NewPaMessagePageState;
+  dispatch: React.Dispatch<NewPaMessagePageReducerAction>;
+}
```

Comment:
**quibble:** I would personally prefer if these components were explicit about what data from `NewPaMesagePageState` they needed and if they were also passed explicit callbacks that were implemented using the `dispatch` function in the component that renders them

Reply:
Honestly, my reason for going with a reducer is a little silly so I agree that explicit data and callbacks in props is the cleaner way to handle this.